packages/next-codemod/bin/upgrade.ts

```diff
@@ -172,12 +172,24 @@ export async function runUpgrade(
 
   fs.writeFileSync(appPackageJsonPath, JSON.stringify(appPackageJson, null, 2))
 
-  const nextDependency = `next@${targetNextVersion}`
-  const dependenciesToInstall = [
-    `react@${targetReactVersion}`,
-    `react-dom@${targetReactVersion}`,
-  ]
+  const dependenciesToInstall = []
   const devDependenciesToInstall = []
+
+  const corePackageNameVersionMapping = {
+    react: `react@${targetReactVersion}`,
+    'react-dom': `react-dom@${targetReactVersion}`,
+    next: `next@${targetNextVersion}`,
+  }
+  for (const packageName of Object.keys(corePackageNameVersionMapping)) {
+    // First check if it's dev dependencies, then add into dev deps;
+    // If it's not, fallback into dependencies.
+    if (appPackageJson.devDependencies?.[packageName]) {
+      devDependenciesToInstall.push(corePackageNameVersionMapping[packageName])
+    } else if (appPackageJson.dependencies?.[packageName]) {
+      dependenciesToInstall.push(corePackageNameVersionMapping[packageName])
+    }
```

Comment:
It's not falling back at the moment
```suggestion
    // If it's not, fallback into dependencies.
    if (appPackageJson.devDependencies?.[packageName]) {
      devDependenciesToInstall.push(corePackageNameVersionMapping[packageName])
    } else {
      dependenciesToInstall.push(corePackageNameVersionMapping[packageName])
    }
```